vignettes/sir_models.Rmd

```diff
@@ -0,0 +1,680 @@
+    exp_noise <- 1e6
+  } else {
+    exp_noise <- pars$compare$exp_noise
+  }
+  
+  incidence_modelled <-
+     prev_state[2, , drop = TRUE] - state[2, , drop = TRUE]
+  incidence_observed <- observed$cases
+  lambda <- incidence_modelled +
+    rexp(n = length(incidence_modelled), rate = exp_noise)
+  dpois(x = incidence_observed, lambda = lambda, log = TRUE)
+}
+```
+
+## Inferring parameters
+
+Using these pieces, we can set up a particle filter as follows:
+
+```{r}
+n_particles <- 100
+filter <- particle_filter$new(data = sir_data,
+                              model = gen_sir,
+                              n_particles = n_particles,
+                              compare = case_compare,
+                              n_threads = 4L,
+                              seed = 1L)
+```
+
+We can now run the particle filter forward, which will run and resample 100 trajectories, and return the final likelihood. It is important to set the correct time-step $dt$ here, as we are using 0.25, rather than the default value of 1 defined in the odin model.
+
+```{r}
+filter$run(save_history = TRUE, pars = list(dt = dt))
+```
+
+If we plot these along with the data, we can see compared to above that only trajectories consistent with the data are kept, and the variance between particles has been reduced:
+
+```{r fig.height=5, fig.width=7}
+history <- filter$history()
+
+par(mar = c(4.1, 5.1, 0.5, 0.5), las = 1)
+matplot(incidence_data$day, t(history[2, , -1]), type = "l",
+        xlab = "Time", ylab = "Number of individuals",
+        col = "#999966", lty = 1, ylim = range(history))
+matlines(incidence_data$day, t(history[3, , -1]), col = "#8c8cd9", lty = 1)
+matlines(incidence_data$day, t(history[4, , -1]), col = "#cc0044", lty = 1)
+matpoints(incidence_data$day, t(true_history[2:4, , -1]), pch = 19,
+          col = c("#999966", "#8c8cd9", "#cc0044"))
+legend("left", lwd = 1, col = c("#999966", "#8c8cd9", "#cc0044"), 
+       legend = c("S", "I", "R"), bty = "n")
+```
+
+Doing the same with a $\beta$ and $\gamma$ which are further from the truth yields a lower likelihood:
+
+```{r}
+filter$run(save_history = TRUE, pars = list(dt = dt, beta = 0.4, gamma = 0.2))
+```
+
+
+```{r fig.height=5, fig.width=7}
+history <- filter$history()
+matplot(incidence_data$day, t(history[2, , -1]), type = "l",
+        xlab = "Time", ylab = "Number of individuals",
+        col = "#999966", lty = 1, ylim = range(history))
+matlines(incidence_data$day, t(history[3, , -1]), col = "#8c8cd9", lty = 1)
+matlines(incidence_data$day, t(history[4, , -1]), col = "#cc0044", lty = 1)
+matpoints(incidence_data$day, t(true_history[2:4, , -1]), pch = 19,
+          col = c("#999966", "#8c8cd9", "#cc0044"))
+legend("left", lwd = 1, col = c("#999966", "#8c8cd9", "#cc0044"), 
+       legend = c("S", "I", "R"), bty = "n")
+```
+
+Typically you will not use the `run()` function of the particle filter directly, and instead use the functions below to infer parameters.
```

Comment:
If the user was developing a new inference method, doing a sensitivity analysis, or manually doing a grid search they might?